_sub/compute/eks-addons/main.tf

```diff
@@ -23,6 +23,21 @@ resource "null_resource" "coredns" {
   depends_on = [null_resource.kubeproxy, local.coredns_version]
 }
 
+# For future EKS upgrades, check the upgrade guide for changes to system:coredns clusterrole,
+# and reflect those changes in the YAML file below.
+resource "null_resource" "coredns_clusterrole" {
```

Comment:
This should not be a null resource but regular kubernetes provider resource, which should be possible with clusterroles or a kubectl resource.